budgetportal/summaries.py

```diff
@@ -553,3 +557,53 @@ def get_aggregate_drilldowns(self):
             openspending_api.get_programme_name_ref(),
             openspending_api.get_econ_class_4_ref(),
         ]
+
+
+class InYearSpending(DepartmentBudgetData):
+    def get_dataset(self):
+        return get_in_year_spending_dataset(
+            self.department.government.sphere.financial_year.slug
+        )
+
+    def get_aggregate_cuts(self):
+        cuts = []
+        openspending_api = self.get_openspending_api()
+        if openspending_api is not None:
+            department_ref = openspending_api.get_department_name_ref()
+            cuts = [
+                department_ref + ":" + "{}".format(self.department.name),
+            ]
+        return cuts
+
+    def get_aggregate_drilldowns(self):
+        openspending_api = self.get_openspending_api()
+        if openspending_api is not None:
+            return [
+                openspending_api.get_programme_name_ref(),
+                openspending_api.get_department_name_ref(),
+                openspending_api.get_subprogramme_name_ref(),
+                openspending_api.get_econ_class_1_ref(),
+                openspending_api.get_econ_class_2_ref(),
+                openspending_api.get_econ_class_3_ref(),
+                openspending_api.get_econ_class_4_ref(),
+            ]
+        else:
+            return None
```

Comment:
we only want these drilldowns in the csv download. We don't want these drilldowns in the chart data. get_aggregate_url is the URL used in the chart, and get_detail_csv_url is the URL with more detail for download. Other charts use all drilldowns to get all the detail, but this dataset in this chart will have too much detail, which is why we should override get_detail_csv_url to use only these drilldowns.

Reply:
- don't override get_detail_csv_url - the csv download will use this via base class' get_detail_aggregate_url call
- return empty list of drilldowns in get_aggregate_drilldowns
- don't override get_aggregate_url - the chart will use this
- override get_detail_aggregate_url to use the custom list of drilldowns

Reply:
updated the code.

> return empty list of drilldowns in get_aggregate_drilldowns
instead of an empty list I had to return `None`. otherwise openspending returns an error.

I had to override `get_detail_csv_url` and `get_aggregate_url` to add an extra `None` check. Because in our case if the dataset does not exist in CKAN, `self.get_openspending_api()` returns `None`. The default functions throw error in these cases. But I also updated `get_detail_csv_url()` so it calls `get_detail_aggregate_url()` now instead of `get_aggregate_url()`.

Now the chart calls `get_aggregate_url()` function which has no drilldowns. Csv calls `get_detail_aggregate_url()` function which has a custom list of drilldowns